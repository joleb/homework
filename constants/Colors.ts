const tintColorLight = "#0a7ea4";
const tintColorDark = "#fff";

export type AvailableThemedColors = keyof typeof lightColors;

enum SupportedThemes {
  light = "light",
  dark = "dark",
}

const lightColors = {
  text: "#11181C",
  background: "#ffffff",
  tint: tintColorLight,
  icon: "#687076",
  tabIconDefault: "#687076",
  tabIconSelected: tintColorLight,
  separator: "#ccc",
  tagBorder: "#000",
  primary: "#6200ee",
  primaryBorder: "#3700b3",
  primaryBackground: "#bb86fc",
  secondary: "#03dac6",
  secondaryBorder: "#018786",
  secondaryBackground: "#03dac6",
  success: "#4caf50",
  successBorder: "#388e3c",
  successBackground: "#c8e6c9",
  warning: "#ff9800",
  warningBorder: "#f57c00",
  warningBackground: "#ffe0b2",
  error: "#f44336",
  errorBorder: "#d32f2f",
  errorBackground: "#ffcdd2",
  cardBackground: "#f5f5f5",
  cardShadow: "#e0e0e0",
};

const darkColors = {
  text: "#ECEDEE",
  background: "#121212",
  tint: tintColorDark,
  icon: "#9BA1A6",
  tabIconDefault: "#9BA1A6",
  tabIconSelected: tintColorDark,
  separator: "#2D3336",
  tagBorder: "#ccc",
  primary: "#bb86fc",
  primaryBorder: "#3700b3",
  primaryBackground: "#6200ee",
  secondary: "#03dac6",
  secondaryBorder: "#018786",
  secondaryBackground: "#03dac6",
  success: "#4caf50",
  successBorder: "#388e3c",
  successBackground: "#c8e6c9",
  warning: "#ff9800",
  warningBorder: "#f57c00",
  warningBackground: "#ffe0b2",
  error: "#cf6679",
  errorBorder: "#b00020",
  errorBackground: "#ffccd5",
  cardBackground: "#1e1e1e",
  cardShadow: "#000000",
};

export const Colors: Record<
  SupportedThemes,
  Record<AvailableThemedColors, string>
> = {
  light: lightColors,
  dark: darkColors,
} as const;

export type ThemedColors = typeof Colors.light | typeof Colors.dark;
