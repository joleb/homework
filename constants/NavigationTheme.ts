import { DefaultTheme, DarkTheme } from "@react-navigation/native";

export const lightNavigationTheme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    primary: "#3B82F6", // Brighter primary color for active navigation items
    background: "#ffffff", // White background for the app
    card: "#FAFAFA", // Softer gray background for navigation cards
    text: "#11181C", // Darker text color for readability
    border: "#E0E0E0", // Light gray border for subtle separation
    notification: "#F59E0B", // Yellow-orange for notifications
  },
};

export const darkNavigationTheme = {
  ...DarkTheme,
  colors: {
    ...DarkTheme.colors,
    primary: "#bb86fc", // Light purple for active navigation items
    background: "#121212", // Dark background for the app
    card: "#1E1E1E", // Dark gray background for navigation cards
    text: "#ECEDEE", // Light text color for readability
    border: "#3E4C59", // Darker border for subtle separation
    notification: "#F59E0B", // Yellow-orange for notifications
  },
};
