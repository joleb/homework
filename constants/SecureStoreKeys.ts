const SecureStoreKeys = {
  password: "PASSWORD",
  email: "EMAIL",
  accessToken: "ACCESS_TOKEN",
} as const;

export default SecureStoreKeys;
