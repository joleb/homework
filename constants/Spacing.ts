const Spacing = {
  xS: 4,
  s: 8,
  xM: 12,
  m: 16,
  l: 24,
  xL: 32,
  xxL: 40,
} as const;

export default Spacing;
