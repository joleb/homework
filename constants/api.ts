export const API_URL = 'https://staging.api.constellation.academy/api/graphql';