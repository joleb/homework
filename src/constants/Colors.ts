export type AvailableThemedColors = keyof typeof lightColors;

export enum SupportedThemes {
  light = "light",
  dark = "dark",
}

export const lightColors = {
  text: "#11181C",
  textError: "#d32f2f",
  background: "#ffffff",
  tint: "#6200EE",
  border: "#9BA1A6",
  icon: "#9BA1A6",
  tabIconDefault: "#9BA1A6",
  tabIconSelected: "#6200EE",
  separator: "#E0E0E0",
  tagBorder: "#333",
  primary: "#3B82F6",
  primaryBorder: "#1D4ED8",
  primaryBackground: "#E3F2FD",
  secondary: "#EF4444",
  secondaryBorder: "#B91C1C",
  secondaryBackground: "#FEE2E2",
  success: "#10B981",
  successBorder: "#047857",
  successBackground: "#D1FAE5",
  warning: "#F59E0B",
  warningBorder: "#B45309",
  warningBackground: "#FEF3C7",
  error: "#EF4444",
  errorBorder: "#B91C1C",
  errorBackground: "#FEE2E2",
  cardBackground: "#FAFAFA",
  cardShadow: "#00000033",
  buttonPrimaryBackground: "#3B82F6",
  buttonPrimaryText: "#FFFFFF",
  buttonSecondaryBackground: "#10B981",
  buttonSecondaryText: "#FFFFFF",
  textInputBackground: "#FFFFFF",
  textInputPlaceholder: "#6B7280",
};

export const darkColors = {
  text: "#ECEDEE",
  textError: "#ff8a80",
  background: "#121212",
  tint: "#bb86fc",
  border: "#3E4C59",
  icon: "#9BA1A6",
  tabIconDefault: "#9BA1A6",
  tabIconSelected: "#bb86fc",
  separator: "#2D3336",
  tagBorder: "#9BA1A6",
  primary: "#bb86fc",
  primaryBorder: "#7B52AB",
  primaryBackground: "#322A47",
  secondary: "#ff8a80",
  secondaryBorder: "#B23C17",
  secondaryBackground: "#4A1F1F",
  success: "#03DAC6",
  successBorder: "#018786",
  successBackground: "#003B38",
  warning: "#F59E0B",
  warningBorder: "#B45309",
  warningBackground: "#4A3009",
  error: "#ff8a80",
  errorBorder: "#B23C17",
  errorBackground: "#4A1F1F",
  cardBackground: "#1E1E1E",
  cardShadow: "#00000088",
  buttonPrimaryBackground: "#3700B3",
  buttonPrimaryText: "#FFFFFF",
  buttonSecondaryBackground: "#03DAC6",
  buttonSecondaryText: "#000000",
  textInputBackground: "#1C1C1C",
  textInputPlaceholder: "#9CA3AF",
};

export const Colors: Record<
  SupportedThemes,
  Record<AvailableThemedColors, string>
> = {
  light: lightColors,
  dark: darkColors,
} as const;
