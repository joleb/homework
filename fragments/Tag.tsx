import { ViewProps, StyleSheet } from "react-native";
import { useMemo } from "react";

import { ThemedText } from "@/components/ThemedText";
import {
  CreateStylesColors,
  useThemeAwareStyles,
} from "@/hooks/useThemeAwareStyles";
import Spacing from "@/constants/Spacing";
import { Colors } from "@/constants/Colors";

interface TagProps extends ViewProps {
  label: string;
  variant?: "default" | "warning";
}

const Tag: React.FC<TagProps> = ({ label, variant = "default", style }) => {
  const styles = useThemeAwareStyles(createStyles);
  const flattenedStyles = useMemo(
    () => StyleSheet.flatten([styles.container, styles[variant], style]),
    [styles, variant, style],
  );
  return (
    <ThemedText
      type="caption"
      style={flattenedStyles}
      lightColor={Colors.light.text}
      darkColor={Colors.dark.text}
    >
      {label}
    </ThemedText>
  );
};

export default Tag;

const createStyles = (colors: CreateStylesColors) =>
  StyleSheet.create({
    container: {
      paddingVertical: Spacing.xS,
      paddingHorizontal: Spacing.s,
      borderWidth: 2,
      borderRadius: 8,
      alignSelf: "flex-start",
    },
    default: {
      borderColor: colors.tagBorder,
    },
    warning: {
      borderColor: colors.warningBorder,
    },
  });
