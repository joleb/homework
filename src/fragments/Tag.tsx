import { StyleSheet, TextProps } from "react-native";
import { useMemo } from "react";

import { ThemedText } from "@/src/fragments/ThemedText";
import {
  CreateStylesColors,
  useThemeAwareStyles,
} from "@/src/hooks/useThemeAwareStyles";
import Spacing from "@/src/constants/Spacing";
import { Colors } from "@/src/constants/Colors";

interface TagProps extends TextProps {
  label: string;
  variant?: "default" | "warning";
}

const Tag: React.FC<TagProps> = ({ label, variant = "default", style }) => {
  const styles = useThemeAwareStyles(createStyles);
  const flattenedStyles = useMemo(
    () => StyleSheet.flatten([styles.container, styles[variant], style]),
    [styles, variant, style],
  );
  return (
    <ThemedText
      type="caption"
      style={flattenedStyles}
      lightColor={Colors.light.text}
      darkColor={Colors.dark.text}
    >
      {label}
    </ThemedText>
  );
};

export default Tag;

const createStyles = (colors: CreateStylesColors) =>
  StyleSheet.create({
    container: {
      paddingVertical: Spacing.xS,
      paddingHorizontal: Spacing.s,
      borderWidth: 2,
      borderRadius: 8,
      alignSelf: "flex-start",
    },
    default: {
      borderColor: colors.tagBorder,
    },
    warning: {
      borderColor: colors.warningBorder,
    },
  });
