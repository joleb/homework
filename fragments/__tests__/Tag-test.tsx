import React from "react";

import { render } from "@testing-library/react-native";

import Tag from "../Tag";

import { Colors } from "@/constants/Colors";

describe("Tag Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = render(<Tag label="Test Label" />);
    expect(toJSON).toMatchSnapshot();
  });
  it("renders correctly with default variant", () => {
    const { getByText } = render(<Tag label="Test Label" />);
    const textElement = getByText("Test Label");
    expect(textElement).toBeTruthy();
    expect(textElement.props.style).toMatchObject({
      borderColor: Colors.light.tagBorder,
    });
  });

  it("renders correctly with warning variant", () => {
    const { getByText } = render(
      <Tag label="Warning Label" variant="warning" />,
    );
    const textElement = getByText("Warning Label");
    expect(textElement).toBeTruthy();
    expect(textElement.props.style).toMatchObject({
      borderColor: Colors.light.warningBorder,
    });
  });

  it("applies custom styles", () => {
    const customStyle = { backgroundColor: "red" };
    const { getByText } = render(
      <Tag label="Custom Style" style={customStyle} />,
    );
    const textElement = getByText("Custom Style");
    expect(textElement).toBeTruthy();
    expect(textElement.props.style).toMatchObject(customStyle);
  });
});
