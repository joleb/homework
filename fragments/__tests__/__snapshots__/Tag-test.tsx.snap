// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component should match snapshot 1`] = `[Function]`;
